src/core/Tree/TreeNode.tsx

```diff
@@ -0,0 +1,192 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { CommonProps, getWindow, useTheme } from '../utils';
+import '@itwin/itwinui-css/css/tree.css';
+import { SvgChevronRight } from '@itwin/itwinui-icons-react';
+import { IconButton } from '../Buttons/IconButton';
+import cx from 'classnames';
+import { TreeContext } from './Tree';
+
+export type TreeNodeProps = {
+  /**
+   * Unique id of the node.
+   */
+  nodeId: string;
+  /**
+   * The main text displayed on the node.
+   */
+  label: React.ReactNode;
+  /**
+   * Small note displayed below main label.
+   */
+  sublabel?: React.ReactNode;
+  /**
+   * Icon shown before label and sublabel content.
+   */
+  icon?: JSX.Element;
+  /**
+   * Is TreeNode disabled.
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * Is TreeNode expanded.
+   * @default false
+   */
+  isExpanded?: boolean;
+  /**
+   * Is TreeNode selected.
+   * @default false
+   */
+  isSelected?: boolean;
+  /**
+   * Callback fired when expanding or closing a TreeNode.
+   */
+  onNodeExpanded?: (nodeId: string, expanded: boolean) => void;
```

Comment:
I am just thinking.. As we are not controlling state at all, it means user actually needs to do something to have Tree working, i.e. expanding node. I do not think it might make sense Tree without expansion. 
Following this, shoulld this prop be required then?